pkg/streamingpromql/operators/function_over_instant_vector.go

```diff
@@ -49,7 +58,24 @@ func (m *FunctionOverInstantVector) ExpressionPosition() posrange.PositionRange
 	return m.expressionPosition
 }
 
+func (m *FunctionOverInstantVector) processScalarArgs(ctx context.Context) error {
+	for _, so := range m.ScalarArgs {
+		sd, err := so.GetValues(ctx)
+		if err != nil {
+			return err
+		}
+		m.scalarArgsData = append(m.scalarArgsData, sd)
```

Comment:
We should be able to pre-allocate `scalarArgsData` based on the length of `ScalarArgs` (and skip this entirely if there are no scalar args)